theano/tensor/nnet/ctc_wrapper.c

```diff
@@ -0,0 +1,242 @@
+#section support_code
+
+int ctc_check_result(ctcStatus_t retcode, const char * msg, int * status)
+{
+    if( CTC_STATUS_SUCCESS != retcode )
+    {
+        // Get error message from underlying library
+        const char * ctc_msg = ctcGetStatusString( retcode );
+
+        PyErr_Format( PyExc_RuntimeError,
+                      "%s | CTC library error message: %s",
+                      msg,
+                      ctc_msg );
+        *status = 1;
+    }
+    *status = 0;
```

Comment:
`*status` will always contain 0, as this line is always executed even if we enter in the `if` block above. Also, this function should return an int, but it currently returns nothing.

I think you want to return the status, isn't it ? So, you could remove the `*status` arg, and then return 1 if retcode is not CTC_STATUS_SUCCESS, and 0 otherwise. Or keep `*status` arg and set this function as a void function.